tests/postgres_tests/test_array.py

```diff
@@ -551,6 +552,71 @@ def test_grouping_by_annotations_with_array_field_param(self):
             1,
         )
 
+    def test_filter_by_array_subquery(self):
+        self.assertSequenceEqual(
+            NullableIntegerArrayModel.objects.alias(
+                same_sized_fields=ArraySubquery(
+                    NullableIntegerArrayModel.objects.filter(
+                        field__len=ArrayLenTransform(models.OuterRef('field'))
+                    ).values('field')
+                )
+            ).filter(same_sized_fields__len=2),
+            self.objs[0:2]
+        )
+
+    def test_annotated_array_subquery(self):
+        self.assertSequenceEqual(
+            NullableIntegerArrayModel.objects.annotate(
+                sibling_ids=ArraySubquery(
+                    NullableIntegerArrayModel.objects.exclude(
+                        id=models.OuterRef('id')
+                    ).values('id')
+                )
+            ).get(id=self.objs[0].id).sibling_ids,
```

Comment:
Why not `first()`? :thinking: 
```suggestion
            ).first().sibling_ids,
```